### PR TITLE
feat(MultiSelect): match experimental spec

### LINKED ISSUE #2009: MultiSelect.Filterable invalid: true styling bug.
## Detailed description

> While using `<MultiSelect.Filterable>` component with `invalid={true}` the style of bottom shadow has a bug. Here is couple of screenshots: 

not focused
Here you can see that red box shadow also has blue color over it.

focused
You can see it even better when input field used for filtering inside MultiSelect is focused

That's all because of `bx--text-input` style.
**Normal state:** while `bx--multi-select` has `data-invalid="true"`, `bx--text-input` which is inside still has `box-shadow: 0 1px 0 0 #5a6872;` 
**Focus state:** `box-shadow: 0 2px 0 0 #3d70b2;` for `.bx--text-input:focus` which causes the bug.

## Steps to reproduce the issue

1. Open sandbox - https://xrmwxlyo5w.codesandbox.io/
2. Observe the bug
3. ?????
4. You are awesome

### PR DESCRIPTION
**Pull request by emyarod**
Closes IBM/carbon-components-react#2109

Closes #2009

This PR adds the v10 multiselect and filterable multiselect

Related https://github.com/IBM/carbon-components-react/issues/1611

depends on https://github.com/IBM/carbon-components/pull/2326 and corresponding version release

we should also merge https://github.com/IBM/carbon-components-react/pull/2090 before this PR

#### Changelog

**New**

- storybook knobs to expose all options
- v10 multiselect and filterable multiselect

**Changed**

- align checkbox classes to vanilla

**Removed**

- duplicated/unused props

### CODE REVIEW

**Comment by emyarod** [comment]
since https://github.com/IBM/carbon-components-react/pull/2090 affects the behavior of the component, it might be good to merge that PR in before this one

**Review by dakahn — APPROVED** [review]

**Comment by asudoh** [comment]
Interim comment: `v9` support code is no longer needed, other than that LGTM.

**Comment by emyarod** [comment]
PR requires designer feedback but it seems our netlify integration is still down

File: src/components/MultiSelect/MultiSelect-story.js [hunk 1]
```diff
@@ -20,15 +18,29 @@ import {
 import { withReadme } from 'storybook-readme';
 import readme from './README.md';
 import MultiSelect from '../MultiSelect';
+import { componentsX } from '../../internal/FeatureFlags';
```
Comment (asudoh) [inline src/components/MultiSelect/MultiSelect-story.js]: Just one remainder of `v10` feature flag ref.

Comment (emyarod) [reply src/components/MultiSelect/MultiSelect-story.js]: removed, thanks!

File: src/components/MultiSelect/FilterableMultiSelect.js [hunk 2]
```diff
@@ -253,15 +257,46 @@ export default class FilterableMultiSelect extends React.Component {
...
-        [`${prefix}--list-box--light`]: light,
+        [`${prefix}--multi-select__wrapper--inline`]: inline,
+        [`${prefix}--list-box__wrapper--inline`]: inline,
+        [`${prefix}--multi-select__wrapper--inline--invalid`]:
+          inline && invalid,
+        [`${prefix}--list-box__wrapper--inline--invalid`]: inline && invalid,
       }
     );
-    return (
+    const className = ({ selectedItem }) =>
```
Comment (asudoh) [inline src/components/MultiSelect/FilterableMultiSelect.js]: `className` as a function is not seen in other components. Probably move this to render callback?

**Review by asudoh — COMMENTED** [review]
Thank you for the update! Just another review comment.

File: src/components/MultiSelect/FilterableMultiSelect.js (hunk 2 above)
Comment (emyarod) [reply src/components/MultiSelect/FilterableMultiSelect.js]: this is how I implemented it for our [Dropdown](https://github.com/carbon-design-system/carbon-components-react/pull/2093/files#diff-700df727ae95bd5df9e9b8baa2eb6a1eR158) as well to get the dropdown state from downshift. Should I create a separate PR to change dropdown?

Comment (asudoh) [reply src/components/MultiSelect/FilterableMultiSelect.js]: Yes, it makes sense!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Comment by dakahn** [comment]
Super psyched for this. Thanks for all your hard work @emyarod 🙏

**Pull request merged**